File: frontend/lib/models/drone.dart
class Drone {
  final int? droneId;
  final String nom;
  final double poidsMax;
  final double autonomieMax;
  final double vitesseMax;
  final int niveauSecurite;
  final String statut;
  final String? zoneVolAutorisee;

  Drone({
    this.droneId,
    required this.nom,
    required this.poidsMax,
    required this.autonomieMax,
    required this.vitesseMax,
    required this.niveauSecurite,
    required this.statut,
    this.zoneVolAutorisee,
  });

  Map<String, dynamic> toJson() {
    return {
      'nom': nom,
      'poids_max': poidsMax,
      'autonomie_max': autonomieMax,
      'vitesse_max': vitesseMax,
      'niveau_securite': niveauSecurite,
      'statut': statut,
      if (zoneVolAutorisee != null) 'zone_vol_autorisee': zoneVolAutorisee,
    };
  }

  factory Drone.fromJson(Map<String, dynamic> json) {
    return Drone(
      droneId: json['drone_id'],
      nom: json['nom'],
      poidsMax: (json['poids_max'] as num).toDouble(),
      autonomieMax: (json['autonomie_max'] as num).toDouble(),
      vitesseMax: (json['vitesse_max'] as num).toDouble(),
      niveauSecurite: json['niveau_securite'],
      statut: json['statut'],
      zoneVolAutorisee: json['zone_vol_autorisee'],
    );
  }
}

enum StatutDrone {
  disponible('disponible', 'Disponible'),
  enMission('en_mission', 'En mission'),
  enMaintenance('en_maintenance', 'En maintenance'),
  horsService('hors_service', 'Hors service');

  final String value;
  final String label;
  const StatutDrone(this.value, this.label);
}

File: backend/app/models/drone.py
from sqlalchemy import Column, Integer, String, Float, Text
from sqlalchemy.orm import relationship
from app.database import Base
import enum


class StatutDrone(str, enum.Enum):
    DISPONIBLE = "disponible"
    EN_MISSION = "en_mission"
    EN_MAINTENANCE = "en_maintenance"
    HORS_SERVICE = "hors_service"


class Drone(Base):
    __tablename__ = "drones"

    drone_id = Column(Integer, primary_key=True, index=True)
    nom = Column(String(255), nullable=False)
    poids_max = Column(Float, nullable=False)  # capacité de charge en kg
    autonomie_max = Column(Float, nullable=False)  # en km
    vitesse_max = Column(Float, nullable=False)  # en km/h
    niveau_securite = Column(Integer, nullable=False)  # niveau de sécurité (échelle de 1 à 5)
    statut = Column(String(50), default=StatutDrone.DISPONIBLE.value)
    zone_vol_autorisee = Column(String(255), nullable=True)  # zones où le drone peut voler

    # Relations
    missions = relationship("Mission", back_populates="drone")
    historique_missions = relationship("HistoriqueMission", back_populates="drone")